### DISCUSSION TITLE
Getting 413 from upload endpoint

### DISCUSSION THREAD

**Discussion by wesmc10** [shrinerb/shrine#678] [Q&A]
I'm using the `upload_endpoint` plugin in a Rails app, and have this line `Shrine.plugin :upload_endpoint, max_size: 30 * 1024 * 1024` in my initializer and this `mount Shrine.upload_endpoint(:cache) => '/uploads'` in my routes. I'm also using Uppy on the frontend and have this `.use(XHR, { endpoint: '/uploads' });` in the JS file.

This all works as expected in development, but I've noticed that when trying to upload larger files (roughly 5 MB and above) in a staging or production environment, it fails to upload the file to /uploads and returns a `413 (Request Entity Too Large)` error. Any ideas why this may be?

**Comment by janko** [comment]
Is it possible that there is a request body limit set at the application server level (e.g. nginx)? Shrine should really only return this for files larger than 30 MB in your case.
